src/dispatch/test/service.py

```diff
@@ -87,9 +87,8 @@ def __init__(
         self.pollers: Dict[DispatchID, Poller] = {}
         self.parents: Dict[DispatchID, Poller] = {}
 
-        self.roundtrips: Optional[OrderedDict[DispatchID, List[RoundTrip]]] = None
-        if collect_roundtrips:
-            self.roundtrips = OrderedDict()
+        self.roundtrips: OrderedDict[DispatchID, List[RoundTrip]] = OrderedDict()
+        self.collect_roundtrips = collect_roundtrips
```

Comment:
This simplifies call sites where we accessed the `roundtrips` field since it cannot be `None` anymore.